### PR TITLE
Added widget drawer (no text widget icon yet)

### LINKED ISSUE #93: Widget Drawer UI
**Is your feature request related to a problem? Please describe.**
I want a widget drawer that can hold widget icons.

**Describe the solution you'd like**
A side bar that can be shown or hidden on the left side of the screen with placeholder widget icons.

**Describe alternatives you've considered**
Having a double click menu to insert widgets however this is not as efficient as dragging from a drawer.

**Additional context**
#65 depends on this.

### PR DESCRIPTION
**Pull request by Benjamin-Piper**
Closes #93

## Proposed Changes

A widget drawer that can take children props. Currently has the Calendar, Todo and Habit Tracker widget icons.

Screenshots: 

Can be accessed through `http://localhost:3000/widget-drawer`.

[recording](https://user-images.githubusercontent.com/68324342/111859948-44b31f00-89a9-11eb-84d5-57e95ac026f8.mp4)

Notes:

- All widget icon CSS customisation can be found under `src/components/Widget.css`. Especially the `--iconColour` CSS variable.
- Currently have not refined the animation. Had to do some tricks as there is currently no way to animate `display: none`.
- Currently using a dummy container in `App.js` to get the formatting working.

### CODE REVIEW

File: telos-frontend/src/components/calendar/WidgetCalendar.jsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+const WidgetTodo = () => {
+  const today = moment();
+  const thisMonth = today.format('MMMM');
+  const thisDay = today.format('DD');
+  return (
+    <div className="widget-icon vertical">
+      <div className="small-month">{thisMonth}</div>
+      <div className="big-number">{thisDay}</div>
+    </div>
+  )
```
Comment (milkyway404) [inline telos-frontend/src/components/calendar/WidgetCalendar.jsx]: Please fix lint issues

File: telos-frontend/src/components/habit-tracker/WidgetHabitTracker.css [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  height: var(--size);
+  width: var(--size);
+  border: 2px solid var(--iconColour);
+  border-radius: 50%;
+  margin-left: 2px;
+}
+
+.circle.filled {
+  background-color: var(--iconColour);
+}
```
Comment (milkyway404) [inline telos-frontend/src/components/habit-tracker/WidgetHabitTracker.css]: no newline at end of file

File: telos-frontend/src/components/habit-tracker/WidgetHabitTracker.jsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import '../Widget.css';
+import './WidgetHabitTracker.css';
+
+const WidgetHabitTracker = () =>
```
Comment (milkyway404) [inline telos-frontend/src/components/habit-tracker/WidgetHabitTracker.jsx]: fix lint

File: telos-frontend/src/components/todo/WidgetTodo.jsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import '../Widget.css';
+
+const WidgetTodo = () =>
```
Comment (milkyway404) [inline telos-frontend/src/components/todo/WidgetTodo.jsx]: fix lint

**Review by milkyway404 — CHANGES_REQUESTED** [review]
Looks really good, just some lint issues that you can fix by running `npm run lint`

Also, is there a issue for fixing the icons later on since (at least the calendar one) was designed to be dynamic? If it isn't, I could re-design it.

File: telos-frontend/src/components/todo/WidgetTodo.jsx (hunk 4 above)
Comment (Benjamin-Piper) [reply telos-frontend/src/components/todo/WidgetTodo.jsx]: ran lint again

File: telos-frontend/src/components/habit-tracker/WidgetHabitTracker.jsx (hunk 3 above)
Comment (Benjamin-Piper) [reply telos-frontend/src/components/habit-tracker/WidgetHabitTracker.jsx]: ran lint again

File: telos-frontend/src/components/habit-tracker/WidgetHabitTracker.css (hunk 2 above)
Comment (Benjamin-Piper) [reply telos-frontend/src/components/habit-tracker/WidgetHabitTracker.css]: ran lint again

File: telos-frontend/src/components/calendar/WidgetCalendar.jsx (hunk 1 above)
Comment (Benjamin-Piper) [reply telos-frontend/src/components/calendar/WidgetCalendar.jsx]: ran lint again

**Comment by Benjamin-Piper** [comment]
> Also, is there a issue for fixing the icons later on since (at least the calendar one) was designed to be dynamic? If it isn't, I could re-design it.

@Milk-Yan Wdym by this? Is it relating to the date changing? If so, I just added it because it looks nice.

**Review by milkyway404 — APPROVED** [review]

File: telos-frontend/src/components/widget-drawer/WidgetDrawer.jsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import './WidgetDrawer.css';
+import { useState } from 'react';
+
+const Arrow = () => (
+  <svg xmlns="http://www.w3.org/2000/svg" height="24" viewBox="0 0 24 24" width="24">
```
Comment (lhuijser) [inline telos-frontend/src/components/widget-drawer/WidgetDrawer.jsx:5]: Is there a material-ui icon that could be used instead of this?

**Review by lhuijser — APPROVED** [review]
Looks good to me! Just one non-urgent question, it doesn't affect much but could make the code shorter. I am super happy with this though, thanks for implementing it :)

**Comment by ODE-BOY** [comment]
Nothing to add that hasn't already been said. Good Job.

**Review by ODE-BOY — COMMENTED** [review]
Nothing more to add that hasn't already been said. It all looks good to me. nice work.

File: telos-frontend/src/components/widget-drawer/WidgetDrawer.jsx (hunk 5 above)
Comment (Benjamin-Piper) [reply telos-frontend/src/components/widget-drawer/WidgetDrawer.jsx:5]: There is. It's called Chevron Left. https://material-ui.com/components/material-icons/ I'll go make a low-priority issue.

**Pull request merged**